### ISSUE TITLE
Rename instantValidation

### ISSUE THREAD

**Issue by bookercodes** [25th-floor/revalidation#30]
Maybe a better name for `instantValidation` would be `validateOnUpdate` (to match `updateValue`), what do you think? 

Actually, what I would personally like to do is rename `updateValue` to `onChange` and then `instantValidation` to `validateOnChange`. 

I like `onChange` more than `updateValue` more because it matches the React prop, and makes it clear that I am delegating the normal event handling to this lib.

**Comment by busypeoples** [comment]
Yes, makes sense. We should definitely rename `updateValue` to `onChange`.
Regarding `validateOnChange` also makes a lot of sense.
